### ISSUE TITLE
VST plugin settings sometimes not displayed correctly in REAPER

### ISSUE THREAD

**Issue by JamesPeters667** [sadko4u/lsp-plugins#101]
I use Xubuntu 19.10, Reaper for Linux, and the latest LSP VST Plugins 1.1.19. I noticed this with the previous VST version 1.1.17 as well.

If I save a preset using Reaper's own effects browser window, the LSP plugin will recall the settings correctly (confirmed by a listening test), however it won't display all the settings correctly until I close the plugin window and then open it again.

If I save a CFG file from the plugin, and use that CFG file to load a preset, the plugin immediately shows all controls as they should be.

Let me know if you'd like me to report this to Reaper's developers (if you suspect this is a Reaper issue).

**Comment by sadko4u** [comment]
I need to reproduce it first.
Since I'm not well familiar with Reaper, could you please provide a short video how to reproduce the case?

**Comment by JamesPeters667** [comment]
Here is a quick video demonstrating the issue. 

https://youtu.be/Ak3xe-e4WLo

There are 2 instances of Compressor Stereo, on 2 tracks. 

I loaded a CFG file into the 1st instance (which is a preset I'd made and saved earlier). Then I saved that preset into Reaper's own VST effects preset system. 

(I had to then close the 2nd plugin instance so that I could find that newly created VST Reaper preset in the list, then I re-opened the 2nd plugin instance.) 

Then in the 2nd plugin instance, I loaded the Reaper VST preset file. The controls of the 2nd instance didn't show the proper saved settings (you can see them compared to those of the 1st plugin instance).

I then closed and re-open that 2nd instance, and the settings displayed correctly (as you can see compared with those of the 1st plugin instance above it).

Earlier today I asked another Reaper for Linux user who uses LSP Plugins if he experienced the same issue, and he confirmed it.

**Comment by sadko4u** [comment]
Thank you for the video, I'll eplore what's happening soon.

**Comment by sadko4u** [comment]
Reproduced the problem.
Created **github-issue-101** branch and pushed the fix: 27853122befef01ce74150ac5ca76fff612d8f84

@JamesPeters667 are you available to test it?

**Comment by JamesPeters667** [comment]
Sorry I'm having trouble building; it's failing on "make install" step. I'm new to this. I don't know how to get the following dependencies: 

gcc-c++ >= 4.7
libgcc_s1 >= 5.2
ladspa-devel >= 1.13

I think I have the rest installed. It's difficult for me to understand what packages these are or what packages they are contained within, since they are named differently in the repos (or possibly not there in the Ubuntu repo).

I don't think I need the ladspa package anyway since I chose to only build the vst module.

**Comment by sadko4u** [comment]
To avoid conflicts you need to uninstall lsp-plugins package installed from the repository first.

Then, for Ubuntu you need to issue the following commands:
```
make clean
sudo apt-get install g++ make ladspa-sdk lv2-dev libjack-jackd2-dev libcairo2-dev libsndfile1-dev  mesa-common-dev php-cli
make
sudo make install
```

I believe I haven't missed something in package list. Otherwise you'll get compilation error.

**Comment by JamesPeters667** [comment]
Thank you for the information. I was missing php-cli even though I had the php package installed. Knowing the specific package names helps a lot for someone like me. 

My previous install was just the .so files from SourceForge, and my VST directory is non-standard, so that wasn't a problem. I did remove them anyway.

I tested a few plugins doing the same test as before. **The UI refreshes properly now** when loading VST presets from Reaper's effects browser. Thank you! I'll have that other Reaper user test this version as well just in case.

Thank you for these plugins by the way. They're great. :)

**Comment by sadko4u** [comment]
Thanks for confirming that the problem has gone.
Merged changes into **devel** branch: f9333f23de438c2e38b62fb8b391cfaba8d32b9d
Removed **github-issue-101** branch.

**Comment by sadko4u** [comment]
Committed change to the **hotfix** branch: 227f53e03b08a133de13cda03bc69d5e28f254e0

**Comment by sadko4u** [comment]
Will close the issue after release

**Comment by stevenjaycohen** [comment]
Ocenaudio shows this behavior as well. Plugins work fine but start with the UI collapsed. Once you grab a corner and resize the window, everything works as expected.

You might want to include it in the list of hosts that support your VSTs too.
https://ocenaudio.com

**Comment by sadko4u** [comment]
@stevenjaycohen 
Seems that the host has some issues with effEditGetRect and audioMasterSizeWindow.

**Comment by stevenjaycohen** [comment]
@sadko4u sending a link to this issue to the developers

**Comment by sadko4u** [comment]
Changes available in 1.1.21 release!
